lib/adal-angular.js

```diff
@@ -127,15 +127,19 @@
                                 event.preventDefault();
                                 // redirect to login start page
                                 if (!_adal.popUp) {
-                                    var loginStartPage = _adal._getItem(_adal.CONSTANTS.STORAGE.LOGIN_REQUEST);
-                                    if (loginStartPage) {
-                                        // prevent the current location change and redirect the user back to the login start page
-                                        _adal.verbose('Redirecting to start page: ' + loginStartPage);
-                                        if (!$location.$$html5 && loginStartPage.indexOf('#') > -1) {
-                                            $location.url(loginStartPage.substring(loginStartPage.indexOf('#') + 1));
+                                    if (_adal.config.navigateToLoginRequestUrl) {
+                                        var loginStartPage = _adal._getItem(_adal.CONSTANTS.STORAGE.LOGIN_REQUEST);
+                                        if (loginStartPage) {
+                                            // prevent the current location change and redirect the user back to the login start page
+                                            _adal.verbose('Redirecting to start page: ' + loginStartPage);
+                                            if (!$location.$$html5 && loginStartPage.indexOf('#') > -1) {
+                                                $location.url(loginStartPage.substring(loginStartPage.indexOf('#') + 1));
+                                            }
+                                            $window.location = loginStartPage;
                                         }
-                                        $window.location = loginStartPage;
                                     }
+                                    else
+                                        $window.location.hash = '';
```

Comment:
I remember some users reported that this caused a reload for them. Can you please what happens when we reset only the hash?